### ISSUE TITLE
readproto occasionally returns a few garbage values

### ISSUE THREAD

**Issue by chipbuster** [JuliaIO/ProtoBuf.jl#164]
I'm attempting to read the internal representation of a GZIP file generated by another program into Julia using protobufs. This is my first project using any ProtoBuf machinery, so it's possible that I've done something very stupid.

I can write and read the ProtoBuf file in my other program without problems, but when I attempt to read the file using ProtoBuf.jl, I occasionally get nonsense numbers. Here's a snippet from an attempt to read ~500 protobuf message files:

```
output_10_0.2_5_unsorted_2AA_.protos: output_10_0.9_4_unsorted_3AA_.protos: Invalid dist codepoint key 2147512584;
Invalid dist codepoint key 1828736520;
output_10_0.9_5_sorted_2AA_.protos: Invalid dist codepoint key 1762132490;
output_30_0.8_2_unsorted_2AA_.protos: Invalid dist codepoint key 1560805898;
output_30_0.9_1_unsorted_2AA_.protos: Invalid dist codepoint key 1275593226;
output_30_0.9_3_sorted_2AA_.protos: Invalid dist codepoint key 1460142602;
output_30_0.9_4_sorted_2AA_.protos: Invalid dist codepoint key 621281802;
output_30_0.9_4_unsorted_2AA_.protos: Invalid dist codepoint key 2667584520;Invalid dist codepoint key 104200;
output_30_0.9_5_sorted_2AA_.protos: Invalid dist codepoint key 1442859016;
output_30_0.9_5_unsorted_2AA_.protos: Invalid dist codepoint key 101188106
```

The keys in question should never be greater than 29, but they're taking on what look like nonsense 32-bit integer values. I have verified that I can read the files fine in both my other program and by dumping them with `protoc --decode`, so I don't believe the files are corrupted. These numbers also change run-to-run.

The protobuf structure is deeply nested, so it's possible that I'm somehow accidentally bypassing some part of ProtoBuf.jl's internal machinery, but I'm not sure how I'd be doing that. I never access the structures with functions aside from `getproperty` and `hasproperty`. In addition, `isinitialized` tells me that the structure is fully init-ed.

I am using threads, but the problem persists with or without them (in fact, the error list with `JULIA_NUM_THREADS=1` is almost twice as long as the one here, which was made with 4 threads).

This is with v0.10.0, though downgrading to v0.9.1 did not seem to fix the issue.

Below is the proto specification for the message. I can also provide the code I'm using to check these values and a few examples of files that give me trouble, if needed.

```
syntax = "proto3";

package DEFLATEProto;

message Literal {
    uint32 value = 1;
}

message Backref {
    uint32 length = 1;
    uint32 distance = 2;
}

message OffsetBackref {
    uint32 offset = 1;
    uint32 length = 2;
    uint32 distance = 3;
}

message DeflateSym {
    oneof sym {
        Literal lit = 1;
        Backref backref = 2;
        OffsetBackref offset = 3;
    }
}

message UncompressedBlock {
    bytes data = 1;
}

// Ideally we would like to store codes directly, but storing variable-length
// bitvectors in map values is hard. Instead, store the length of the code and
// use the canonical encoding algorithms to reconstruct the actual codes.
message CompressedBlock {
    map<uint32, uint32> lenlit_codelen = 3;
    map<uint32, uint32> dist_codelen = 4;
    repeated DeflateSym data = 1;
}

message UnderlyingBlock {
    oneof data {
        UncompressedBlock raw = 10;
        CompressedBlock block = 1;
    }
}

message DEFLATEBlock {
    bool bfinal = 1;
    UnderlyingBlock data = 2;
}

message DEFLATEStream {
    repeated DEFLATEBlock blocks = 1;
}
```

**Comment by tanmaykm** [comment]
There doesn't seem anything unusual in the proto file. It will be useful if you could send a minimal working example to reproduce this.

**Comment by chipbuster** [comment]
@tanmaykm I'm sorry for the long delay. I tried looking into this at the end of February--unfortunately, it seems that I was working on a dirty copy of my code repository at the time and did not correctly commit the version of the code that was causing my problem. I tried to restore all the commits I'd made in the week leading up to my filing this report, but I was unable to reproduce this bug on any of them. (I also had several updates of julia and other system libraries from my system package manager in the meantime, so I can't rule out that there was some sort of packaging error occurring on Arch--such things are known to happen occasionally).

I was hoping to be able to dedicate some time to reproduce this, but it's increasingly looking like that's not going to happen. Apologies for the noise here--I'll make sure to save off a tarball or git tag next time I file reports so that I always have reproducing code.
